server/item/stack.go

```diff
@@ -42,6 +44,23 @@ func NewStack(t world.Item, count int) Stack {
 	return Stack{item: t, count: count, id: newID()}
 }
 
+// WithArmourTrim returns a new stack with the ArmorTrim passed.
+// This only applies if the stack is type Armour.
+func (s Stack) WithArmourTrim(trim ArmourTrim) Stack {
+	if _, ok := s.item.(Armour); !ok {
+		return s
+	}
+
+	s.armorTrim = &trim
+	return s
+}
+
+// ArmourTrim returns the ArmourTrim.
+// if this returns nil it does not have an armor trim, or it is not an armour
+func (s Stack) ArmourTrim() *ArmourTrim {
+	return s.armorTrim
+}
```

Comment:
Try and avoid exposing pointers in cases like this, you already do it in the WithArmourTrim() method, but I'd just adapt ArmourTrim() to return (ArmourTrim, bool) to stop the user having to handle nil pointers